### PR TITLE
Vectorized Transform Spline node

### PR DESCRIPTION
**Pull request by harisreedhar**
Vectorized all inputs of ``Transform Spline`` node

### CODE REVIEW

File: animation_nodes/nodes/spline/transform_spline.py [hunk 1]
```diff
@@ -1,22 +1,63 @@
...
     bl_label = "Transform Spline"
-    codeEffects = [VectorizedSocket.CodeEffect]
 
     useSplineList: VectorizedSocket.newProperty()
+    useMatrixList: VectorizedSocket.newProperty()
 
     def create(self):
         socket = self.newInput(VectorizedSocket("Spline", "useSplineList",
             ("Spline", "spline"), ("Splines", "splines")))
         socket.dataIsModified = True
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/spline/transform_spline.py:16]: `dataIsModified` is only needed when `useMatrixList == False`.

File: animation_nodes/nodes/spline/transform_spline.py [hunk 2]
```diff
@@ -1,22 +1,63 @@
...
+        return outSplines
+
+    def execute_MultipleSpline_SingleMatrix(self, splines, matrix):
+        for spline in splines:
+            spline.transform(matrix)
+        return splines
+
+    def execute_MultipleSpline_MultipleMatrix(self, splines, matrices):
+        _splines = VirtualPyList.create(splines, Spline())
+        _matrices = VirtualPyList.create(list(matrices), Matrix())
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/spline/transform_spline.py:56]: `VirtualMatrix4x4List`.

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]

File: animation_nodes/nodes/spline/transform_spline.py (hunk 1 above)
Comment (harisreedhar) [reply animation_nodes/nodes/spline/transform_spline.py]: Done

File: animation_nodes/nodes/spline/transform_spline.py (hunk 2 above)
Comment (harisreedhar) [reply animation_nodes/nodes/spline/transform_spline.py]: Done

**Review by OmarEmaraDev — APPROVED** [review]
Thanks!

**Pull request merged**